src/draw/DefaultDrawer.js

```diff
@@ -61,479 +90,455 @@ class DefaultDrawer {
+    return p.matrixTransform(svg.getScreenCTM().inverse());
   }
 
   /**
-   * Draws all Links.
-   * @param {ComponentLink[]} links - List of links you want to draw.
-   * @return {Boolean} Returns true if normally execute.
+   * Convert svg coordinates into screen coordinates
+   * @param svgX
+   * @param svgY
+   * @param svg
+   * @return {DOMPoint}
+   * @constructor
    */
-  drawLinks(links) {
-    if (!links) return;
-
-    if (document.querySelector('.links') === null) {
-      this.d3.select(`#${this.rootId}`)
-        .append('g')
-        .classed('links', true);
-    }
 
-    const linkGen = this.d3.linkHorizontal()
-      .source((data) => this.getAnchorPosition(data).source)
-      .target((data) => this.getAnchorPosition(data).target);
-
-    this.d3.select('.links')
-      .selectAll('.link')
-      .data(links)
-      .join('path')
-      .classed('link', true)
-      .attr('d', linkGen)
-      .attr('fill', 'none')
-      .attr('stroke', 'black')
-      .attr('stroke-width', 2);
+  SVGToScreen(svgX, svgY,svg = null) {
+    svg = svg ?? this.svg.node()
+    const p = svg.createSVGPoint()
+    p.x = svgX
+    p.y = svgY
+    return p.matrixTransform(svg.getScreenCTM());
   }
 
   /**
-   * Draws all Components in the parentId Element.
-   * @param {Component[]} components - List of components you want to draw.
-   * @param {String} parentId - Id of the container where you want to draw.
-   * @param {ComponentLink[]} links - List of links you want to draw.
-   * @return {Boolean} Returns true if normally execute.
+   * Find a component in the given components tree
+   * @param rootNode - The starting point of the search
+   * @param componentId - The id to search for
+   * @return {null|Component} - The component or null if not found
    */
-  drawComponents(components, parentId, links, linksDefinition) {
-    const d3Container = this.d3.select(parentId === this.rootId ? `#${parentId}` : `#${parentId} .component-container`)
-      .selectAll(`.${parentId}.component`)
-      .data(components, (data) => data.id);
 
-    const d3Elements = d3Container
-      .enter().append('svg');
+  findComponentById(rootNode,componentId) {
+    let res = null
+    if (rootNode) {
+      if (rootNode.id === componentId) res = rootNode
 
-    d3Elements
-      .attr('id', (data) => data.id)
-      .attr('class', (data) => `${parentId} ${data.definition.type} component component-${data.definition.model} ${data.definition.isContainer ? 'isContainer' : ''}`)
-      .html((data) => this.resources.models[data.definition.model]);
-
-    this.__drawModels(components, parentId);
-
-    components.forEach((component) => {
-      this.setSelectionAction(component, components, links);
-      if (component.children.length > 0) {
-        this.drawComponents(component.children, component.id, links, linksDefinition);
+      if (!res && rootNode.children) {
+        res = rootNode.children.map(c => this.findComponentById(c, componentId)).filter(c=>c)[0] ?? null
```

Comment:
improve readability and indentation

Reply:
missing ;

Reply:
done, should be replaced with similar method from PluginData further down the line